src/index.ts

```diff
@@ -108,8 +108,7 @@ const sqlTypeToTypescriptType = (
 			const referencedInterface = modelNameToCamelCaseName(
 				m.tables[f.references!.resourceName].name,
 			);
-			const nullable = f.required ? '' : '?';
-			return `{ __id: ${referencedDataType} } | [${referencedInterface}${nullable}]`;
+			return `{ __id: ${referencedDataType}${f.required ? '' : ' | null'} } | [${referencedInterface}${f.required ? '' : '?'}]`;
```

Comment:
@Page- My understanding here is that you are saying that the current generated type, for example, on balena-model `Application` generates `should_be_running__release: { __id: number } | [Release?] | null;`. Creating an empty release I see that this type is currenttly correct as I get a `null` release on an empty fleet.

With this fix the typying would become `should_be_running__release: { __id: number | null } | [Release?];` but I don't see where the response actually changes?